ui/apps/platform/cypress/integration/credentialExpiry.test.js

```diff
@@ -1,41 +1,34 @@
 import dateFns from 'date-fns';
 import withAuth from '../helpers/basicAuth';
-import * as api from '../constants/apiEndpoints';
 import { selectors } from '../constants/CertExpiration';
-import { visitMainDashboard } from '../helpers/main';
-
-describe('Cert Expiration Banner', () => {
+import {
+    interactAndWaitForCentralCertificateDownload,
+    interactAndWaitForScannerCertificateDownload,
+    renderCentralCredentialExpiryBanner,
+    renderScannerCredentialExpiryBanner,
+} from '../helpers/credentialExpiry';
+
+describe('Credential expiry', () => {
     withAuth();
 
-    const mockCertExpiryAndVisitHomepage = (endpoint, expiry) => {
-        cy.intercept('GET', endpoint, {
-            body: { expiry },
-        }).as('certExpiry');
-        visitMainDashboard();
-        cy.wait('@certExpiry');
-    };
-
-    describe('Central', () => {
+    describe('for central', () => {
         it('should not display banner if central cert is expiring more than 14 days later', () => {
             const expiry = dateFns.addHours(dateFns.addDays(new Date(), 15), 1);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
+
+            renderCentralCredentialExpiryBanner(expiry);
```

Comment:
One thought on naming. At first look, I thought this function might have manipulated the DOM in some way due to the `render` prefix but it turns out it uses the standard `intercept/wait` cycle to mock the request and cause the banner to render. Any reason not to go with something like `visitWithCentralCredentialExpiryBanner`?

Reply:
Good call.

Because visit functions without direct object in name have path as first argument, I used a more verbose alternative `visitSystemConfigurationWithCentralCredentialExpiryBanner` although which page is an implementation detail.